src/liquid_parquet/src/liquid_array/raw/fsst_array.rs

```diff
@@ -426,6 +432,160 @@ impl FsstArray {
+        }
+
+        // Flush the buffer to disk
+        buffer
+            .flush()
+            .map_err(|e| Error::other(format!("Failed to flush buffer: {e}")))?;
+
+        Ok(())
+    }
+
+    /// Loads the symbol table from a file and rebuilds the compressor.
+    /// Uses the same format as the one saved by `save_symbol_table()`.
+    pub fn load_symbol_table(file_path: &str) -> Result<Compressor> {
+        // Open and read the file
+        let mut file = File::open(file_path)
+            .map_err(|e| Error::other(format!("Failed to open file {file_path}: {e}")))?;
+
+        // Read the whole file
+        let mut buffer = Vec::new();
+        file.read_to_end(&mut buffer)
+            .map_err(|e| Error::other(format!("Failed to read file {file_path}: {e}")))?;
+
+        // Validate the buffer length
+        if buffer.is_empty() {
+            return Err(Error::new(ErrorKind::InvalidData, "File is empty"));
+        }
+
+        // Read the length of the symbol table
+        let symbols_len = usize::from(buffer[0]);
+
+        // Validate we have enough data for the symbol lengths
+        if buffer.len() < 1 + symbols_len {
+            return Err(Error::new(
+                ErrorKind::InvalidData,
+                "File is malformed: insufficient data for symbol length",
+            ));
+        }
+
+        // Read the lengths of each symbol
+        let symbols_lengths = &buffer[1..1 + symbols_len];
+
+        let symbols_start = 1 + symbols_len;
+        let symbols_end = symbols_start + symbols_len * SYMBOL_SIZE_BYTES;
+
+        if buffer.len() < symbols_end {
+            return Err(Error::new(
+                ErrorKind::InvalidData,
+                "File is malformed: insufficient data for symbols",
+            ));
+        }
+
+        let symbols_bytes_slice = &buffer[symbols_start..symbols_end];
+
+        let mut loaded_symbols: Vec<Symbol> = Vec::with_capacity(symbols_len);
+
+        // Loop over the byte slice in chunks and build Symbols
+        for chunk in symbols_bytes_slice.chunks_exact(SYMBOL_SIZE_BYTES) {
+            // chunk is &[u8] of length `SYMBOL_SIZE_BYTES`
+            let symbol_bytes: [u8; SYMBOL_SIZE_BYTES] = chunk
+                .try_into()
+                .expect("Chunk should be {SYMBOL_SIZE_BYTES} bytes");
+            loaded_symbols.push(Symbol::from_slice(&symbol_bytes));
+        }
+
+        // Check if we read the expected number of symbols
+        if loaded_symbols.len() != symbols_len {
+            return Err(Error::new(
+                ErrorKind::InvalidData,
+                format!(
+                    "File is malformed: read {} symbols, expected {}",
+                    loaded_symbols.len(),
+                    symbols_len
+                ),
+            ));
+        }
+
+        // Call rebuild_from and handle panics
+        let compressor =
+            std::panic::catch_unwind(|| Compressor::rebuild_from(loaded_symbols, symbols_lengths))
```

Comment:
I think here we should call the function as is, even if it may panic. We already checked the preconditions, if it still errors, it's a non-recoverable bug